financial-connections/src/test/java/com/stripe/android/financialconnections/features/networkinglinksignup/LinkSignupHandlerForInstantDebitsTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.stripe.android.financialconnections.features.networkinglinksignup
+
+import com.stripe.android.financialconnections.ApiKeyFixtures.consumerSessionSignup
+import com.stripe.android.financialconnections.ApiKeyFixtures.sessionManifest
+import com.stripe.android.financialconnections.ApiKeyFixtures.syncResponse
+import com.stripe.android.financialconnections.domain.AttachConsumerToLinkAccountSession
+import com.stripe.android.financialconnections.domain.GetOrFetchSync
+import com.stripe.android.financialconnections.domain.HandleError
+import com.stripe.android.financialconnections.features.networkinglinksignup.NetworkingLinkSignupState.Payload
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.Pane
+import com.stripe.android.financialconnections.navigation.NavigationManager
+import com.stripe.android.financialconnections.presentation.Async
+import com.stripe.android.financialconnections.repository.FinancialConnectionsConsumerSessionRepository
+import com.stripe.android.financialconnections.utils.TestIntegrityRequestManager
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class LinkSignupHandlerForInstantDebitsTest {
+
+    private lateinit var handler: LinkSignupHandlerForInstantDebits
+    private val consumerRepository = mock<FinancialConnectionsConsumerSessionRepository>()
+    private val getOrFetchSync = mock<GetOrFetchSync>()
+    private val attachConsumerToLinkAccountSession = mock<AttachConsumerToLinkAccountSession>()
+    private val integrityRequestManager = TestIntegrityRequestManager()
+    private val navigationManager = mock<NavigationManager>()
+    private val handleError = mock<HandleError>()
+
+    @Before
+    fun setUp() {
+        handler = LinkSignupHandlerForInstantDebits(
+            consumerRepository,
+            attachConsumerToLinkAccountSession,
+            integrityRequestManager,
+            getOrFetchSync,
+            navigationManager,
+            "applicationId",
+            handleError
+        )
+    }
+
+    private val validPayload = Payload(
+        merchantName = "Mock Merchant",
+        emailController = mock(),
+        prefilledEmail = "test@stripe.com",
+        sessionId = "fcsess_1234",
+        appVerificationEnabled = false,
+        phoneController = mock {
+            whenever(it.getCountryCode()).thenReturn("US")
+        },
+        isInstantDebits = true,
+        content = mock()
+    )
+
+    @Test
+    fun `performSignup should navigate to next pane on success`() = runTest {
```

Comment:
Could you also add a test for the non-verified case?

Reply:
done! also added a test For the Networking Handler.